src/Todos.tsx

```diff
@@ -0,0 +1,286 @@
+import React, { useState, useEffect, useMemo } from 'react';
+import classNames from 'classnames';
+import { Todo } from './types/Todo';
+import {
+  getTodos,
+  addTodos,
+  deleteTodo,
+  updateTodo,
+} from './api/todos';
+import { TodoItem } from './components/TodoItem/TodoItem';
+
+enum ErorrType {
+  NONE,
+  ADDERORR,
+  DELETEERORR,
+  UPDATEERORR,
+}
+
+enum Filter {
+  ALL = 'all',
+  ACTIVE = 'active',
+  COMPLETED = 'completed',
+}
+
+interface User {
+  id:number,
+  name:string,
+  email:string,
+}
+
+type TodosProps = {
+  user:User;
+};
+
+export const Todos: React.FC<TodosProps> = ({ user }) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [erorr, setErorr] = useState<ErorrType>(ErorrType.NONE);
+  const [filterTodos, setFilterTodos] = useState(Filter.ALL);
+  const [title, setTitle] = useState('');
+  const [isDisabled, setIsDisabled] = useState(false);
+
+  let timeErorr: number;
+
+  const funcVisibleTodos = (arrItems: Todo[], typeFilter: Filter) => {
+    const items = [...arrItems];
+
+    return items.filter(item => {
+      switch (typeFilter) {
+        case Filter.ALL:
+          return item;
+
+        case Filter.ACTIVE:
+          return item.completed === false;
+
+        case Filter.COMPLETED:
+          return item.completed === true;
+
+        default:
+          return item;
+      }
+    });
+  };
+
+  const visibleTodos = useMemo(() => {
+    return funcVisibleTodos(todos, filterTodos);
+  }, [todos, filterTodos]);
+
+  const ativeTodos = todos.filter(todo => todo.completed === false);
+  const completedTodos = todos.filter(todo => todo.completed === true);
+
+  const clearErorr = () => {
+    setErorr(ErorrType.NONE);
+    window.clearTimeout(timeErorr);
```

Comment:
it's better to split this big component into a few smaller ones:
you may create a separate component for error and handle it in useEffect (it will look a bit clearer)